packages/repository/src/repositories/constraint-utils.ts

```diff
@@ -55,12 +55,15 @@ export function constrainDataObject<T extends Entity>(
 ): DataObject<T> {
   const constrainedData = cloneDeep(originalData);
   for (const c in constraint) {
-    if (constrainedData.hasOwnProperty(c))
+    if (constrainedData.hasOwnProperty(c)) {
+      if (originalData[c] === constraint[c]) continue;
```

Comment:
`===` won't work for objects such as `ObjectId`, but I can live with the limitation. Please add a comment.

Reply:
Good catch! I'll add a comment.